e2e/addObservationWithoutData.e2e.js

```diff
@@ -0,0 +1,36 @@
+import {
+  by,
+  device,
+  element,
+  expect,
+  waitFor
+} from "detox";
+
+describe( "Add observation without evidence", () => {
+  beforeAll( async () => {
+    await device.launchApp( { permissions: { location: "always" } } );
+  } );
+
+  beforeEach( async () => {
+    await device.reloadReactNative();
+  } );
+
+  it( "should open app with the observation list screen", async () => {
+    const loginText = element( by.text( "Log in to iNaturalist" ) );
+    await waitFor( loginText ).toBeVisible().withTimeout( 2000 );
+    await expect( loginText ).toBeVisible();
+  } );
+
+  it( "should navigate to observation add screen on add evidence button pressed", async () => {
+    await waitFor( element( by.text( "Log in to iNaturalist" ) ) )
+      .toBeVisible()
+      .withTimeout( 2000 );
+    await element( by.id( "camera-options-button" ) ).tap();
```

Comment:
Let's use `by.label()` here and use accessibility labels for these buttons. In theory we won't be writing many e2e tests, but I think that's a good habit for us to have so we don't forget to add these labels.

Reply:
I was starting to use `by.label()` but had an error that it is matching multiple elements. It is checking against the native view hierarchy and not the RN side, so I am not sure there is a way around that problem. I checked the Detox [documentation](https://wix.github.io/Detox/docs/api/matchers/#bylabellabel) and they actually recommend to match elements by unique identifiers. So, for now I changed all the elements I matched by text to match by id as well. Then we can use the unit and integration tests to get into the habit of using accessibility labels.

Reply:
I think that's a fine approach for situations where there really are multiple elements in view with exactly the same accessibility text... but I think those cases are going to be rare, and this specific case isn't one of them. I've got this specific test working with `by.label( )` so I'll commit that as a proof-of-concept.